Yh ropes save image link

JackHuang313 opened the pull request:
Closes #2

JackHuang313 commented on .rubocop.yml:
```diff
@@ -6,6 +6,7 @@ AllCops:
     - 'vendor/**/*'
     - 'node_modules/**/*'
     - 'bin/*'
+    - 'config/initializers/*'
```
The reason for adding this line of code is to stop rubocop from reporting offenses as follow:
```
config/initializers/simple_form.rb:2:1: C: Layout/EmptyLineAfterMagicComment: Add an empty line after magic comments.
#
^
config/initializers/simple_form.rb:17:5: C: Layout/AlignHash: Align the elements of a hash literal if they span more than one line.
    hint_class: :field_with_hint, error_class: :field_with_errors, valid_class: :field_without_errors do |b|
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
which are kind of unsolvable.

JackHuang313 commented on app/models/image.rb:
```diff
@@ -0,0 +1,4 @@
+class Image < ApplicationRecord
+  validates :link, presence: true
+  validates :link, url: true
```
Currently, the validate_url gem is imported for url validation. After the skeleton of the ropes program is completed, I will probably launch a separate PR to make the url validation more accurate and robust.

JackHuang313 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Ropes Program #2 - Save an image link](https://trello.com/c/7S6LnWhZ/797-ropes-program-2-save-an-image-link)

Santhanaraman commented on app/models/image.rb:
```diff
@@ -0,0 +1,4 @@
+class Image < ApplicationRecord
+  validates :link, presence: true
+  validates :link, url: true
```
Minor: I think you can have those two lines together as a single line, like
`validates :link, presence: true, url: true`

JackHuang313 replied:
Modified in https://github.com/AppFolioOnboarding/image-sharer-JackHuang313/pull/14/commits/0d74f63b69d222737acbba07e6f4dc9c1f7c268f

Santhanaraman commented on db/migrate/20190213190128_create_images.rb:
```diff
@@ -0,0 +1,9 @@
+      t.timestamps
+    end
+  end
+end
```
You don't have to change it here, but it is better to have migration as a separate commit

Santhanaraman commented on config/routes.rb:
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
   root 'application#home'
+  resources :images
```
Since we only need `new`, `create`, and `show` can we use `only` or `except` to restrict the routes? https://guides.rubyonrails.org/routing.html#restricting-the-routes-created

JackHuang313 replied:
Modified in https://github.com/AppFolioOnboarding/image-sharer-JackHuang313/pull/14/commits/0d74f63b69d222737acbba07e6f4dc9c1f7c268f

Santhanaraman commented on app/controllers/images_controller.rb:
```diff
@@ -0,0 +1,25 @@
+  def create
+    @image = Image.new(image_params)
+
+    if @image.save
```
I guess this is fine but an alternative way could be to use `@image.save!` (this raises exception when the validation fails) and in controller we can rescue the exception and handle it.

JackHuang313 commented:
> Always have a meaningful commits even while addressing code review comments, [0d74f63](https://github.com/AppFolioOnboarding/image-sharer-JackHuang313/commit/0d74f63b69d222737acbba07e6f4dc9c1f7c268f) can be replace with 2 separate atomic commits. Its fine for now, so am approving this PR

Thanks for mentioning that! I am still getting used to making atomic commits and I will keep it in mind.
